elastic/kibana
# [App Search] Added the Documents View

## Summary

This PR adds a Documents View for the App Search Plugin, accessible via the 'Documents' link in the Engines sub nav.

The view is ported from our stand-alone UI. Most notable about this view is that it uses the https://github.com/elastic/search-ui library.

It looks like it balloons the size of the `enterpriseSearch` async chunk to ~1.3MB, increasing by ~630.3KB (based on the build report below). I'm guessing this is due to all of the transitive dependencies brought in by @elastic/react-search-ui. Is there a way to see a breakdown of the chunk?

We don't use the majority of the transitive dependencies that are brought in. I thought these might get shaken out by tree shaking, but perhaps not. We might be able to do some work on search-ui to omit these dependencies.

Given that, I'm looking for thoughts and opinions on how to proceed. The commit that adds Search UI is here: 9b3a7a677c9569fbaeb9b3c4f0388add452443c9. CC specifically @spalger and @constancecchen.

The view is completely functional, but I do leave a few things out to be implemented later as separate PRs:

What it includes:
- Added `@elastic/react-search-ui` and `@elastic/search-ui-app-search-connector` as dependencies.
- Added the search-ui-based search experience to the Documents page.

What it does not include the following, which will be implemented in separate PRs:
- This PR does not include the correct "Result" view, which must requires the `StuiResult` component to be migrated. This uses a temporary view for showing results.
- This PR does not include the facets and the ability to configure those facets.
- This PR does not include polling to detect when indexing finishes on an Engine.
- This PR does not have the Index Documents button wired up.

![screencapture-localhost-5601-app-enterprise-search-app-search-engines-national-parks-demo-documents-2020-11-20-12_17_40](https://user-images.githubusercontent.com/1427475/99829555-6f3fb900-2b2a-11eb-93d9-d88102c2c242.png)

**QA**

You can see various forms of "No Results" messages in this PR by using the following scenarios:
- Perform a search that returns no results on the Documents page
- Visit the Documents page on a regular engine that has no Documents indexed
- Visit the Document page on a *meta* engine that has no Documents indexed

Another scenario to note:
- When visiting this page for a meta engine, there should be a blue banner displayed at the top

**Reviewing this PR**

⚠️ I recommend stepping through this commit by commit. ⚠️ 

It is a big PR, but the atomic commits should make this easier to review in logical chunks.

Note that the majority of the new code is more or less copy / paste from the old code repository. The changes I made were mostly limited to adding `i18n`, fixing axe failures, and writing tests. I made little to no intentional refactors.

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
